src/application.rs

```diff
@@ -99,4 +99,24 @@ impl Application for RusticApp {
 
         Ok(())
     }
+
+    /// Shut down this application gracefully
+    fn shutdown(&self, shutdown: Shutdown) -> ! {
+        let exit_code = match shutdown {
+            Shutdown::Crash => 1,
+            _ => 0,
+        };
+        self.shutdown_with_exitcode(shutdown, exit_code)
+    }
+}
+
+impl RusticApp {
+    /// Shut down this application gracefully, exiting with given exit code.
+    fn shutdown_with_exitcode(&self, shutdown: Shutdown, exit_code: i32) -> ! {
+        if let Err(e) = self.state().components().shutdown(self, shutdown) {
+            fatal_error(self, &e)
+        }
+
+        process::exit(exit_code);
+    }
```

Comment:
I wonder if there is a value of adding this method instead of in-lining this code into the Application::shutdown implementation.

Reply:
Yes, you are right. Currently only `Application::shutdown` is used. However, in future we plan to add more exit codes than just 0 or 1. 

E.g. a `rustic check` which found no error (caveat: if errors are found there will be no error code 1 unless https://github.com/rustic-rs/rustic_core/pull/224 has been merged) but only warning.

Or a `rustic backup` which had problems with some files, but could in general backup most files.

For this and because of https://github.com/iqlusioninc/abscissa/pull/865/files (which unfortunately is still not released), I added this method...

Reply:
I understand better!
IMO, the fact that we're waiting for this abscissa pull request to be released could be specified in a comment?

Whether or not you add the comment, I'm okay with resolving my comment/merging.